source/How-To-Guides/Launch-file-different-formats.rst

```diff
@@ -370,9 +679,9 @@ Python, XML, or YAML: Which should I use?
 
 For most applications the choice of which ROS 2 launch format comes down to developer preference.
 However, if your launch file requires flexibility that you cannot achieve with XML or YAML, you can use Python to write your launch file.
-Using Python for ROS 2 launch is more flexible because of following two reasons:
+Using Python for ROS 2 launch is more flexible because:
 
 * Python is a scripting language, and thus you can leverage the language and its libraries in your launch files.
-* `ros2/launch <https://github.com/ros2/launch>`_ (general launch features) and `ros2/launch_ros <https://github.com/ros2/launch_ros>`_ (ROS 2 specific launch features) are written in Python and thus you have lower level access to launch features that may not be exposed by XML and YAML.
+* `ros2/launch <https://github.com/ros2/launch>`_ (general launch features) and `ros2/launch_ros <https://github.com/ros2/launch_ros>`_ (ROS 2 specific launch features) are written in Python and thus you have lower level access to launch features that may not be yet exposed via XML and YAML.
```

Comment:
What is the difference between `launch` and `launch_ros`?  Why have 2 APIs?  How do they work together / complement each other?

Reply:
To answer this here (the answer I believe belongs to the packages' documentation), `launch` was designed to be extensible. `launch_ros` uses leverages that to introduce ROS 2 specific actions, of which there are none in core `launch` . Why? I believe it was mix of a sense of architectural cleanliness (ie. separation of concerns) and a desire to have `launch` be ROS 2 agnostic (I vaguely recall `launch` being used for Gazebo sim).